torch/utils/cpp_extension.py

```diff
@@ -714,6 +714,7 @@ def append_std17_if_no_std_present(cflags) -> None:
         def unix_cuda_flags(cflags):
             cflags = (COMMON_NVCC_FLAGS +
                       ['--compiler-options', "'-fPIC'"] +
+                      ['--compiler-options', "'--fdiagnostics-color=auto'"] +
```

Comment:
You should only add this one if you are running the process in terminal, in CI/logged into some sort of system it will look bad. See other examples throughout the code how to detect if ones stdout is terminal

Reply:
Done! It's interesting that this is the case that was precisely my understanding of what the auto flag was for